modules/heat_conduction/src/actions/MortarGapHeatTransferAction.C

```diff
@@ -0,0 +1,326 @@
+  const int lm_order = primal_type.order.get_order();
+
+  if (primal_type.family != LAGRANGE)
+    mooseError("The mortar thermal action can only be used with LAGRANGE finite elements");
+
+  params.set<MooseEnum>("family") = Utility::enum_to_string<FEFamily>(primal_type.family);
+  params.set<MooseEnum>("order") = Utility::enum_to_string<Order>(OrderWrapper{lm_order});
+
+  if (_user_provided_mortar_meshes)
+    params.set<std::vector<SubdomainName>>("block") = {
+        getParam<SubdomainName>("secondary_subdomain")};
+  else
+    params.set<std::vector<SubdomainName>>("block") = {action_name + "_secondary_subdomain"};
+
+  params.set<std::vector<Real>>("scaling") = {getParam<Real>("thermal_lm_scaling")};
+  auto fe_type = AddVariableAction::feType(params);
+  auto var_type = AddVariableAction::variableType(fe_type);
+
+  _problem->addVariable(var_type, action_name + "_thermal_lm", params);
+}
+
+void
+MortarGapHeatTransferAction::addConstraints()
+{
+  checkForExistingSubdomains();
+
+  InputParameters params = _factory.getValidParams("ModularGapConductanceConstraint");
+  const std::string action_name = MooseUtils::shortName(name());
+
+  params.applyParameters(parameters());
+  params.set<bool>("correct_edge_dropping") = true;
+  params.set<bool>("use_displaced_mesh") = true;
+
+  params.set<BoundaryName>("primary_boundary") = getParam<BoundaryName>("primary_boundary");
+  params.set<BoundaryName>("secondary_boundary") = getParam<BoundaryName>("secondary_boundary");
+
+  if (_user_provided_mortar_meshes)
+  {
+    params.set<SubdomainName>("primary_subdomain") = getParam<SubdomainName>("primary_subdomain");
+    params.set<SubdomainName>("secondary_subdomain") =
+        getParam<SubdomainName>("secondary_subdomain");
+  }
+  else
+  {
+    params.set<SubdomainName>("primary_subdomain") = action_name + "_primary_subdomain";
+    params.set<SubdomainName>("secondary_subdomain") = action_name + "_secondary_subdomain";
+  }
+
+  params.set<NonlinearVariableName>("variable") = action_name + "_thermal_lm";
+
+  params.set<VariableName>("secondary_variable") =
+      getParam<std::vector<VariableName>>("temperature")[0];
+
+  const std::vector<UserObjectName> user_object_vector =
+      getParam<std::vector<UserObjectName>>("user_object_physics");
+
+  std::vector<UserObjectName> uoname_strings(0);
+  unsigned int conduction_index = 0;
+  unsigned int radiation_index = 0;
+
+  for (const auto i : index_range(user_object_vector))
+  {
+    if (user_object_vector[i] == "GapFluxModelConduction")
+      uoname_strings.push_back("gap_flux_model_conduction_object_" + MooseUtils::shortName(name()) +
+                               "_" + Moose::stringify(conduction_index++));
+    else if (user_object_vector[i] == "GapFluxModelRadiation")
+      uoname_strings.push_back("gap_flux_model_radiation_object_" + MooseUtils::shortName(name()) +
+                               "_" + Moose::stringify(radiation_index++));
+  }
+
```

Comment:
```suggestion
  for (const auto & uo: user_object_vector)
  {
    if (uo == "GapFluxModelConduction")
      uoname_strings.push_back("gap_flux_model_conduction_object_" + MooseUtils::shortName(name()) +
                               "_" + Moose::stringify(conduction_index++));
    else if (uo == "GapFluxModelRadiation")
      uoname_strings.push_back("gap_flux_model_radiation_object_" + MooseUtils::shortName(name()) +
                               "_" + Moose::stringify(radiation_index++));
  }
```

I don't understand this design. The parameter you get `user_object_vector` from is of type `UserObjectName`, which is the block name a specific UO appears under in the input file. But here you compare it to a _type name_?

Reply:
>which is the block name a specific UO appears under in the input file

I guess I am using it as the object name, i.e. defining the physics to be added.

Reply:
It should be a multimooseenum then. The current way breaks input file validation.

Reply:
What kind of validation?

Reply:
> input file validation

Reply:
More specifically, the NEAMS workbench input file validation effort. Input parameter types have a meaning. `UserObjectName` means that valid values here must be subblocks of the `[UserObjects]` block. You have exactly two valid options for `user_object_physics`. Why not use a parameter type that communicates that to the user and enables both autocompletion and input file validation?